frontend/public/components/instantiate-template.tsx

```diff
@@ -129,55 +129,39 @@ const stateToProps = (state: RootState) => ({
   models: state.k8s.getIn(['RESOURCES', 'models']),
 });
 
-class TemplateForm_ extends React.Component<
-  TemplateFormProps & WithTranslation,
-  TemplateFormState
-> {
-  constructor(props: TemplateFormProps & WithTranslation) {
-    super(props);
+const TemplateForm_: React.FC<TemplateFormProps> = (props) => {
+  const { preselectedNamespace: ns = '', activePerspective, perspectiveExtensions, obj } = props;
 
-    const { preselectedNamespace: namespace = '' } = this.props;
-    const parameters = this.getParameterValues(props);
-    this.state = {
-      namespace,
-      parameters,
-      inProgress: false,
-      error: '',
-    };
-  }
-
-  componentDidUpdate(prevProps: TemplateFormProps & WithTranslation) {
-    if (this.props.obj.data?.parameters !== prevProps.obj.data?.parameters) {
-      const parameters = this.getParameterValues(this.props);
-      this.setState({ parameters });
-    }
-  }
-
-  getParameterValues = (props: TemplateFormProps & WithTranslation) => {
-    const templateParameters: TemplateParameter[] = props.obj.data.parameters || [];
+  const getParameterValues = (input: TemplateFormProps) => {
+    const templateParameters: TemplateParameter[] = input.obj.data.parameters || [];
     return templateParameters.reduce((acc, { name, value }: TemplateParameter) => {
       acc[name] = value;
       return acc;
     }, {});
   };
 
-  onNamespaceChange = (namespace: string) => {
-    this.setState({ namespace });
-  };
+  const [prevProps, setPrevProps] = React.useState(props);
+  const [namespace, setNamespace] = React.useState(ns);
+  const [parameters, setParameters] = React.useState(getParameterValues(props));
+  const [inProgress, setInProgress] = React.useState(false);
+  const [error, setError] = React.useState('');
+
+  const { t } = useTranslation();
+
+  React.useEffect(() => {
+    if (obj.data?.parameters !== prevProps.obj.data?.parameters) {
+      const params = getParameterValues(props);
+      setParameters(params);
+    }
+    setPrevProps(props);
+  }, [props, obj.data?.parameters, prevProps.obj.data?.parameters]);
```

Comment:
The intent here seems to be "whenever `obj.data.parameters` change, recompute `getParameterValues()` and update the `parameters` state variable".

I think the `props` reference is due to `componentDidUpdate` method contract that works with the whole `props` object, which influenced the way the original code was written.

This could be simplified via memoized value based on the input data which is `obj.data.parameters`

```ts
const parameters = React.useMemo(() => {
  return (obj.data.parameters || []).reduce((acc, { name, value }) => {
    acc[name] = value;
    return acc;
  }, {});
}, [obj]);
```

with this, I think you can remove

```ts
const [prevProps, setPrevProps] = React.useState(props);
const [parameters, setParameters] = React.useState(getParameterValues(props));
```

as well as the `getParameterValues` function declaration itself.

Reply:
I kept parameters as a state value because the onParameterChanged() function also updates parameters, but I simplified the UseEffect() hook and removed prevProps and getParameterValues().